integrations/slack/src/misc/utils.ts

```diff
@@ -131,9 +131,42 @@ export const getSlackTarget = (conversation: Conversation) => {
   return { channel, thread_ts: thread }
 }
 
-export async function sendSlackMessage(botToken: string, ack: AckFunction, payload: ChatPostMessageArguments) {
-  const client = new WebClient(botToken)
-  const response = await client.chat.postMessage(payload)
+const isValidUrl = (str: string) => {
+  try {
+    new URL(str)
+    return true
+  } catch (err) {
+    return false
+  }
+}
+
+const getOptionalProps = (ctx: IntegrationCtx, logger: IntegrationLogger) => {
+  if (ctx.configuration.botAvatarUrl && isValidUrl(ctx.configuration.botAvatarUrl)) {
+    return {
+      username: ctx.configuration.botName,
+      icon_url: ctx.configuration.botAvatarUrl,
+    }
+  }
+
+  if (ctx.configuration.botAvatarUrl && !isValidUrl(ctx.configuration.botAvatarUrl)) {
+    logger.forBot().warn('Invalid bot avatar URL')
+  }
+
+  return {
+    username: ctx.configuration.botName,
```

Comment:
Should we also check if `.botName` is not whitespace (i.e. `.botName.trim() !== ""`) otherwise use `undefined` to prevent whitespace in the bot name causing an error (?) when sending to Slack?

Reply:
good idea, didn't think about it